servers/zms/src/main/java/com/yahoo/athenz/zms/store/AuthHistoryRecord.java

```diff
@@ -0,0 +1,111 @@
+/*
+ *
+ *  * Copyright The Athenz Authors
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.yahoo.athenz.zms.store;
+
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSortKey;
+
+import java.util.Objects;
+
+@DynamoDbBean
```

Comment:
This should be a generic class regardless of where the data is stored. why are we using dynamodb annotations here?

Reply:
The annotations are required for the "mapping table" functionality of the dynamoDB client:
        this.mappedTable = client.table(tableName, TableSchema.fromBean(AuthHistoryRecord.class));
It allows writing queries in a much more convenient way.
I agree that the class should be local to the dynamoDB implementation and the conversion from this structure to AuthHistory.java should take place as part of DynamoDBAuthHistoryStoreConnection and not in the ZMSImpl level.